leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/PresenceServiceImpl.java

```diff
@@ -47,54 +47,119 @@ public void removeListener(PresenceListener listener) {
     }
 
     @Override
-    public boolean isOnline(Registration registration) {
-        return Presence.ONLINE.equals(clientStatus.get(registration.getEndpoint()));
+    public boolean isClientSleeping(Registration registration) {
+        PresenceStatus presenceStatus = clientStatusList.get(registration.getEndpoint());

```

Comment:
This is a public API, so user could use it with a Registration which does not exist anymore, right?
So how does we handle this ?

Personally,  I will go for `isPresent` / `isAwake` / `isOnline` instead of `isClientSleeping`.
and `isPresent` means we have a status for this registration and the status is present.
(About naming I sill feel, that we talk more about presence than "awakeness")

I even ask myself if this could be better to implement sleeping state by "no entry in the map" (but this is maybe too soon to go for this solution)

Reply:
I am using the Q-Mode with actual hardware IoT nodes, so for me it is more intuitive to use the AWAKE/SLEEPING naming. But if you really want it, I can change it to the ONLINE/OFFLINE one.

Apart from that I agree about "isPresent means we have a status for this registration and the status is present".